scalyr_agent/monitors_manager.py

```diff
@@ -208,6 +210,10 @@ def stop_manager(self, wait_on_join=True, join_timeout=5):
 
         start_time = time.time()
 
+        log.info("Stopping shared thread pools")
+        ThreadPoolExecutorFactory.shutdown(wait=False, cancel_futures=True)
```

Comment:
To me it seems like that this should live on the syslog monitor `stop()` method and not inside the MonitorsManager itself.

Reply:
The thread pools are shared across the syslog monitors. That's why it needs to be here.

Reply:
Ah, ok, makes sense, thanks for clarifying.

Could this cause any issues since each monitor is running in a separate thread already?

Reply:
That's a good question.
ThreadPool.submit (this method is thread safe) adds a callable item with parameters to a queue and starts a new thread and adds it to an internal pool if num_threads < self._max_workers.
What I was worried about is creating and starting a long running nested thread - possibly blocking joining of the parent thread. But that's not the case in python.
It would be a better design not to call processing_executor.submit from inside thread from a request_reading_executor pool, but abstractions around BaseServer don't allow it. 
Ideally we should use an async server like tornado to handle these requests, but that's for another PR :)

Reply:
> To me it seems like that this should live on the syslog monitor `stop()` method and not inside the MonitorsManager itself.

Looking more into the code of syslog monitors, I've realized you're right.
Since SyslogMonitor is in charge of starting and stopping all Syslog Servers, it makes sense for it to be responsible for creation and shutting down of the pools.

Reply:
In the end it really needs to be in the Monitor Manager, I understood the Syslog Monitor wrong. It starts servers for one monitor configuration only.